Expense/expenseRouter.js

```diff
@@ -0,0 +1,40 @@
+const express = require("express");
+
+const Expense = require("./Expenses");
+
+const router = express.Router();
+
+router
+  .route("/")
+  .get((req, res) => {
+    Expense.find()
+      .populate("budget")
+      .populate("category")
+      .then(expenses => {
+        res.status(200).json(expenses);
+      })
+      .catch(err => {
+        res.status(500).json({
+          errorMessage: "An error occured while attempting to find expenses",
+          error: err
+        });
+      });
+  })
+
+  .post((req, res) => {
```

Comment:
We're missing a key element here for the post function, and that's saving refs to the Expense instance.

Here's an example of saving refs to Budget and Category to an Expense:

```
.post((req, res) => {
  const { amount, budget, category, description } = req.body

  Budget.findOne({ title: budget }).then(bud => {
    Category.findOne({ title: category }).then(cat => {
      const expense = new Expense({
        amount, budget: bud._id, category: cat._id, description
      });
      expense
        .save()
        .then(expense => res.status(200).json(expense))
        .catch(err => {
          res.status(500).json({
            errorMessage: "An error occured while attempting to save expense",
            error: err
          });
        });
    })
  })
});
```


Make sure your POST `req.body` has:
```
{
    "description": "description",
    "amount": 210,
    "budget": "aaa",
    "category": "Food/Dining"
}
```
And that `'aaa'` and `'Food/Dining'` exist in Budget and Category DB, respectively